### PR TITLE
update documentation

### PR DESCRIPTION
**Pull request by mt-dfrey**
- doc(sdk) add TSDoc comments to public API
- chore(docs) run yarn build:docs
- doc(sdk) replace markdown API doc with links typedoc
- doc(sdk) update readme with badge maintainer instructions

baac479815aea4c2104ce31466e95534f0b6a003 is a large diff because it generates a lot of static pages. You can ignore it for review (as long as you review the previous commit that makes changes to the typescript source files 😉 )

### CODE REVIEW

File: src/typings.ts [hunk 1]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
...
 export interface ConfigsOptions extends PrivateConfigsOptions {
+  /**
+   * Email used to pre-fill the email field in login or sign up or form.
+   *
+   * Set the default value via {@link MtLinkSdk.init}
+   */
   email?: string;
+  /**
+   * A redirection URL for redirecting a guest back to in the following condition:
+   * - Guest clicks on `Back to [App Name]`` button in any Moneytree screen.
```
Comment (mt-rorykelly) [inline src/typings.ts:36]: It looks like there is an extra backtick here.
```suggestion
   * - Guest clicks on `Back to [App Name]` button in any Moneytree screen.
```

File: src/typings.ts [hunk 2]
```diff
@@ -113,7 +251,19 @@ export type AuthorizeUrlOptions = Omit<AuthorizeOptions, 'isNewTab'>;
...
   PrivateParams & {
+    /**
+     * Environment for the SDK to connect to, the SDK will connect to the Moneytree production server by default.
+     * - Moneytree clients should use `staging` for development as `develop` may contain unstable features.
+     * - `local` is only for SDK development as it has local dependencies.
+     */
     mode?: Mode;
+    /**
+     * Force Moneytree to load content in this specific locale. A default value will be auto detected based on guest
+     * langauges configurations and location if available.
```
Comment (mt-rorykelly) [inline src/typings.ts:262]: Small typo `langauges` here
```suggestion
     * languages configurations and location if available.
```

File: src/typings.ts [hunk 3]
```diff
@@ -152,17 +333,26 @@ export interface RequestLoginLinkOptions extends ConfigsOptions {
...
   scope: string;
   client_id: null | string;
+  /** application related information */
   app: null | {
+    /** Application name */
     name: string;
+    /** @hidden */
     is_mt: boolean;
   };
+  /** User related informatino */
```
Comment (mt-rorykelly) [inline src/typings.ts:355]: Small typo `informatino`
```suggestion
  /** User related information */
```

File: src/typings.ts [hunk 4]
```diff
@@ -97,14 +190,59 @@ interface AuthorizeConfigsOptions {
...
+   * - transactions_write
+   * - expense_claims_read
+   * - categories_read
+   * - investment_accounts_read
+   * - investment_transactions_read
+   * - notifications_read
+   * - request_refresh
+   * - life_insurance_read
+   *
+   * See the [LINK Platform documenation](https://docs.link.getmoneytree.com/docs/api-scopes) for more details
```
Comment (mt-ronkorving) [inline src/typings.ts:230]: typo: `documenation`

File: src/typings.ts [hunk 5]
```diff
@@ -113,7 +251,19 @@ export type AuthorizeUrlOptions = Omit<AuthorizeOptions, 'isNewTab'>;
...
   PrivateParams & {
+    /**
+     * Environment for the SDK to connect to, the SDK will connect to the Moneytree production server by default.
+     * - Moneytree clients should use `staging` for development as `develop` may contain unstable features.
+     * - `local` is only for SDK development as it has local dependencies.
+     */
     mode?: Mode;
+    /**
+     * Force Moneytree to load content in this specific locale. A default value will be auto detected based on guest
+     * languages configurations and location if available.
```
Comment (mt-ronkorving) [inline src/typings.ts:262]: configurations plural?

File: src/typings.ts [hunk 6]
```diff
@@ -138,6 +310,15 @@ export type OnboardUrlOptions = Omit<OnboardOptions, 'isNewTab'>;
 
 export type ServiceId = 'vault' | 'myaccount' | 'link-kit';
 
+/**
+ * - `settings` - Main Moneytree account settings screen.
+ * - `settings/authorized-applications` - List of apps currently connected to Moneytree.
+ * - `settings/change-language` - Change Moneytree account language screen.
+ * - `settings/email-preferences` - Change Moneytree email preferences screen
```
Comment (mt-ronkorving) [inline src/typings.ts:317]: Every line "Moneytree account something" except this one.

File: src/typings.ts [hunk 7]
```diff
@@ -138,6 +310,15 @@ export type OnboardUrlOptions = Omit<OnboardOptions, 'isNewTab'>;
 
 export type ServiceId = 'vault' | 'myaccount' | 'link-kit';
 
+/**
+ * - `settings` - Main Moneytree account settings screen.
+ * - `settings/authorized-applications` - List of apps currently connected to Moneytree.
```
Comment (mt-ronkorving) [inline src/typings.ts:315]: suggestion: "to the Moneytree account."?

File: src/typings.ts [hunk 8]
```diff
@@ -152,17 +333,26 @@ export interface RequestLoginLinkOptions extends ConfigsOptions {
 }
 
 export interface TokenInfo {
+  /** token issuer */
   iss: string;
+  /** token cretaion time */
```
Comment (mt-ronkorving) [inline src/typings.ts:338]: typo: cretaion

File: src/typings.ts [hunk 9]
```diff
@@ -152,17 +333,26 @@ export interface RequestLoginLinkOptions extends ConfigsOptions {
...
   iat: number;
+  /** token expiry */
   exp: number;
+  /** token audience(s) */
   aud: string[];
+  /** token subject - the moneytree ide fo the user */
```
Comment (mt-ronkorving) [inline src/typings.ts:344]: typos: "moneytree ide fo"

File: src/typings.ts [hunk 10]
```diff
@@ -175,7 +365,9 @@ export interface Token {
   access_token: string;
   refresh_token: string;
   token_type: string;
+  /** created at in seconds */
```
Comment (mt-ronkorving) [inline src/typings.ts:368]: "in seconds" is cryptic for a date/time value.
perhaps mention that it's a unix timestamp in seconds?

File: src/typings.ts [hunk 11]
```diff
@@ -175,7 +365,9 @@ export interface Token {
...
   created_at: number;
+  /** expiry in seconds */
```
Comment (mt-ronkorving) [inline src/typings.ts:370]: same here

Comment (mt-ronkorving) [inline src/typings.ts:370]: Is that seconds after created_at? If so, maybe we can be explicit about it.

File: src/index.ts [hunk 12]
```diff
@@ -66,37 +90,129 @@ export class MtLinkSdk {
...
     this.storedOptions.samlSubjectId = value;
   }
 
+  /**
+   * OAuth authorization method to request guest consent to access data via the [Link API](https://getmoneytree.com/jp/link/about).
+   *
+   * The only supported flow is authorization code grant with PKCE [PKCE](https://auth0.com/docs/flows/concepts/auth-code-pkce)
+   * If the user is not logged in yet this will show the login screen and redirect the guest to the consent screen after they log in.
+   * After the guest consents, they will be redirected to the redirect URI with an authorization code.
+   * If the user is already logged and has granted consent in the redirection happens immediately.
```
Comment (mt-ronkorving) [inline src/index.ts:110]: Line 109: "guest"
Line 110: "user"

I don't know which word is canonical, but I think we should stick to one.

File: src/typings.ts [hunk 13]
```diff
@@ -1,11 +1,21 @@
...
+/** @hidden */
 export interface PrivateParams {
+  /**
+   * Brand Moneytree apps with client's branding. E.g: logo or theme.
+   * @remarks
+   * This is an internal attribute. Please do not use it unless instructed by your integration representative.
+   */
   cobrandClientId?: string;
+  /**
+   * Sets subject Id for saml session version.
```
Comment (mt-ronkorving) [inline src/typings.ts:13]: ID?

File: src/index.ts [hunk 14]
```diff
@@ -118,6 +234,12 @@ export class MtLinkSdk {
     }
   }
 
+  /**
+   * This method can generate URLs for various services provided by Moneytree, such as Moneytree Account Settings and Vault.
+   *
+   * This API has exactly the same parameters as {@link openService}, the only difference being that it returns an URL
```
Comment (mt-ronkorving) [inline src/index.ts:240]: typo: "it returns"

File: src/index.ts [hunk 15]
```diff
@@ -51,6 +51,30 @@ export class MtLinkSdk {
     mode: 'production'
   };
 
+  /**
+   * Call `init` to initialize the SDK and set default options for API calls.
+   *
+   * Some LINK APIs can be used without calling `init`.
+   * Calls related to OAuth require a client id which can only be set via the `init` function.
```
Comment (mt-ronkorving) [inline src/index.ts:58]: suggestion: ID

File: src/index.ts [hunk 16]
```diff
@@ -66,37 +90,129 @@ export class MtLinkSdk {
...
+   * If the user is not logged in yet this will show the login screen and redirect the guest to the consent screen after they log in.
+   * After the guest consents, they will be redirected to the redirect URI with an authorization code.
+   * If the user is already logged and has granted consent in the redirection happens immediately.
+   *
+   * You can pass the {@link AuthorizeOptions.forceLogout} option to force the guest to log in, even if they have an active session.
+   *
+   * You can also choose to display the sign up form when the user is not logged in yet by setting the {@link AuthorizeOptions.authAction} option.
+   *
+   * @remark
+   * You must initialize the SDK via the {@link init} before calling this API.
```
Comment (mt-ronkorving) [inline src/index.ts:117]: suggestion: remove "the" before `{@link init}`

File: src/index.ts [hunk 17]
```diff
@@ -66,37 +90,129 @@ export class MtLinkSdk {
...
+   */
   public authorize(options?: AuthorizeOptions): void {
     authorize(this.storedOptions, options);
   }
 
+  /**
+   * This method generates an URL for OAuth authorization that requires guest consent to access data via [Link API](https://getmoneytree.com/jp/link/about).
+   *
+   * See {@link authorize} API for authorization details. This API has exactly the same parameters as {@link authorize},
+   * the only difference is that it returns an URL instead of opening immediately with `window.open`.
```
Comment (mt-ronkorving) [inline src/index.ts:132]: typo: "a URL" instead of "an URL"

File: src/index.ts [hunk 18]
```diff
@@ -66,37 +90,129 @@ export class MtLinkSdk {
...
+   */
   public authorizeUrl(options?: AuthorizeUrlOptions): string {
     return authorizeUrl(this.storedOptions, options);
   }
 
+  /**
+   * The onboard API allows a new guest to get onboard faster without having to go through the registration process.
+   * All you have to do is provide an emai address and pass the same options parameter as the {@link authorize} function.
```
Comment (mt-ronkorving) [inline src/index.ts:140]: typo: "emai" -> "email"

File: src/index.ts [hunk 19]
```diff
@@ -66,37 +90,129 @@ export class MtLinkSdk {
...
+   * We will display a consent screen explaining the access requests and applicable scopes.
```
Comment (mt-ronkorving) [inline src/index.ts:141]: Suggestion: "We" -> "Moneytree" Or "Moneytree LINK" or ... something like that?

File: src/index.ts [hunk 20]
```diff
@@ -66,37 +90,129 @@ export class MtLinkSdk {
...
+   * @throws If the `email` property is not set (via {@link init} or in the `options` parameter of this function).
+   */
   public onboard(options?: OnboardOptions): void {
     onboard(this.storedOptions, options);
   }
 
+  /**
+   * This method generates a URL for guest onboarding.
+   *
+   * This API has exactly the same parameters as {@link onboard}, the only difference being that it returns an URL
```
Comment (mt-ronkorving) [inline src/index.ts:165]: "an URL" -> "a URL"

File: src/index.ts [hunk 21]
```diff
@@ -66,37 +90,129 @@ export class MtLinkSdk {
...
+   * Logout current user from Moneytree.
+   */
   public logout(options?: LogoutOptions): void {
     logout(this.storedOptions, options);
   }
 
+  /**
+   * This method generates a URL to log out the guest.
+   *
+   * This API has exactly the same parameters as {@link logout}, the only difference being that it returns an URL
```
Comment (mt-ronkorving) [inline src/index.ts:182]: "an URL" -> "a URL"

File: src/index.ts [hunk 22]
```diff
@@ -66,37 +90,129 @@ export class MtLinkSdk {
...
+   */
   public openService(serviceId: 'myaccount', options?: MyAccountOpenServiceOptions): void;
+  /**
+   * Pass `serviceId: 'vault'` to open the Vault service.
+   *
+   * Depending on the Vault sub-page you want to open, you can pass the following options:
+   * - `serviceList`: opens the vault service list page, pass {@link VaultOpenServiceViewServiceList} as options.
+   * - `serviceConnection`: opens the vault service connection page, pass {@link VaultOpenServiceViewServiceConnection} as options.
+   * - `connectionSetting`: opens the vault connection setting page, pass {@link VaultOpenServiceViewConnectionSetting} as options.
+   * - `customerSupport`: opens the vault customer support page, pass {@link VaultOpenServiceViewCustomerSupport} as options.
```
Comment (mt-ronkorving) [inline src/index.ts:212]: Given that "Vault" is branding, it seems appropriate to always write it capitalized like on lines 206 and 208. So in the description for all these options I would suggest replacing "vault" with "Vault".

File: src/typings.ts [hunk 23]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
 export type AuthnMethod = typeof supportedAuthnMethod[number];
 
 export interface ConfigsOptions extends PrivateConfigsOptions {
+  /**
+   * Email used to pre-fill the email field in login or sign up or form.
```
Comment (mt-ronkorving) [inline src/typings.ts:29]: "Email address used to"?

Typically, this documentation seems to refer to email addresses not as merely "email", but as "email address" (which I'm a fan of, but I believe I've seen a few cases where it just says "email" when referring to an address).

File: src/typings.ts [hunk 24]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
...
+   *
+   * Set the default value via {@link MtLinkSdk.init}
+   */
   email?: string;
+  /**
+   * A redirection URL for redirecting a guest back to in the following condition:
```
Comment (mt-ronkorving) [inline src/typings.ts:35]: "condition" -> "conditions"
or
"condition" -> "one of the following conditions"

File: src/typings.ts [hunk 25]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
...
+   * - Guest clicks on `Back to [App Name]` button in any Moneytree screen.
+   * - Guest refuses to give consent to access permission in the consent screen.
+   * - Guest logs out from Moneytree via an app with this client id
+   * - Revoke an app's consent from settings screen opened via an app with this client id
```
Comment (mt-ronkorving) [inline src/typings.ts:39]: suggest: "id" -> "ID"

also:

Who is the actor in this scenario? All other conditions mention that it's the guest.

File: src/typings.ts [hunk 26]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
...
+   * Email used to pre-fill the email field in login or sign up or form.
+   *
+   * Set the default value via {@link MtLinkSdk.init}
+   */
   email?: string;
+  /**
+   * A redirection URL for redirecting a guest back to in the following condition:
+   * - Guest clicks on `Back to [App Name]` button in any Moneytree screen.
+   * - Guest refuses to give consent to access permission in the consent screen.
+   * - Guest logs out from Moneytree via an app with this client id
```
Comment (mt-ronkorving) [inline src/typings.ts:38]: suggest: "id" -> "ID"

File: src/typings.ts [hunk 27]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
...
+   * - Revoke an app's consent from settings screen opened via an app with this client id
+   *
+   * ⚠️ No `Back to [App Name]` button will be shown if this value is not set, and any of the actions mentioned above will redirect the guest back to login screen by default.
```
Comment (mt-ronkorving) [inline src/typings.ts:41]: "to login screen" -> "to the login screen"

File: src/typings.ts [hunk 28]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
...
+   *
+   * Set the default value via {@link MtLinkSdk.init}
+   */
   backTo?: string;
+  /**
+   * Show login or sign up screen when a session does not exist during an {@link MtLinkSdk.authorize} call.
```
Comment (mt-ronkorving) [inline src/typings.ts:47]: "Show" -> "Show the"

File: src/typings.ts [hunk 29]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
...
+   *
+   * Set default value via {@link MtLinkSdk.init}
+   *
+   * @defaultValue 'login'
+   */
   authAction?: AuthAction;
+  /**
+   * If you wish to disable the login to sign up form toggle button and vice-versa in the auth screen, set this to `false`.
```
Comment (mt-ronkorving) [inline src/typings.ts:55]: I would suggest putting the button description in quotes, like on line 63. Otherwise this line is very hard to parse.

File: src/typings.ts [hunk 30]
```diff
@@ -15,19 +25,71 @@ export const supportedAuthnMethod = ['passwordless', 'sso', 'credentials'] as co
...
+  /**
+   * If you wish to disable the `Stay logged in for 30 days` checkbox in the login screen, set this to `false`.
+   *
+   * Set default value via {@link MtLinkSdk.init}
+   *
+   * @defaultValue true
+   */
   showRememberMe?: boolean;
+  /**
+   * Call method and open/render in a new browser tab, by default all views open in the same tab.
```
Comment (mt-ronkorving) [inline src/typings.ts:71]: Suggestion: ", by" -> ". By"

File: src/typings.ts [hunk 31]
```diff
@@ -97,14 +190,59 @@ interface AuthorizeConfigsOptions {
   forceLogout?: boolean;
 }
 
-interface OAuthSharedParams {
+export interface OAuthSharedParams {
+  /**
+   * The state parameter for OAuth flows, see [here](https://auth0.com/docs/protocols/oauth2/oauth-state) for more details.
+   *
+   * If you generates an identifier for the OAuth authorization on your server make sure to set this value explicitly so
```
Comment (mt-ronkorving) [inline src/typings.ts:197]: typo: "generates" -> "generate"

File: src/typings.ts [hunk 32]
```diff
@@ -97,14 +190,59 @@ interface AuthorizeConfigsOptions {
...
+   * that you can use to acquire the access token after the OAuth redirect occurs.
```
Comment (mt-ronkorving) [inline src/typings.ts:198]: "can use it"?

File: src/typings.ts [hunk 33]
```diff
@@ -113,7 +251,19 @@ export type AuthorizeUrlOptions = Omit<AuthorizeOptions, 'isNewTab'>;
 export type Mode = 'production' | 'staging' | 'develop' | 'local';
 export type InitOptions = Omit<Omit<Omit<AuthorizeOptions, 'forceLogout'>, 'codeChallenge'>, 'pkce'> &
   PrivateParams & {
+    /**
+     * Environment for the SDK to connect to, the SDK will connect to the Moneytree production server by default.
+     * - Moneytree clients should use `staging` for development as `develop` may contain unstable features.
+     * - `local` is only for SDK development as it has local dependencies.
```
Comment (mt-ronkorving) [inline src/typings.ts:257]: Maybe we should mention "development by Moneytree engineers" to be on the safe side? Just a minor suggestion.

File: src/typings.ts (hunk 11 above)
Comment (mt-dfrey) [reply src/typings.ts:370]: I copied the description from [the RFC](https://datatracker.ietf.org/doc/html/rfc6749#section-5.1)

> The lifetime in seconds of the access token.  
> For example, the value "3600" denotes that the access token will expire in one hour from the time the response was generated.

and added a link to the spec on the type.

File: src/index.ts (hunk 12 above)
Comment (mt-dfrey) [reply src/index.ts:110]: I replaced all `guest` with `user` in the docs.
Some of our APIs still have guest (like the scope `guest_read` and some API response objects).
But this is 99% consistent now.

Comment (mt-dfrey) [reply src/index.ts:110]: and yes, logged in is better 😁

File: src/typings.ts (hunk 25 above)
Comment (mt-dfrey) [reply src/typings.ts:39]: also the user, I made it explicit. and changed it to ID.

**Comment by mt-dfrey** [comment]
@mt-ronkorving @mt-rorykelly 
Thanks for your your feedback, I addressed all points.

**Review by mt-ronkorving — APPROVED** [review]

**Review by mt-rorykelly — APPROVED** [review]
